utils/promtool.py

```diff
@@ -0,0 +1,27 @@
+import yaml
+import tempfile
+from subprocess import run, PIPE
+
+
+class PromtoolError(Exception):
+    pass
+
+
+def check_rule(yaml_spec):
+    try:
+        with tempfile.NamedTemporaryFile() as fp:
+            fp.write(yaml.dump(yaml_spec).encode())
+            fp.seek(0)
```

Comment:
why do you need to do this?

Reply:
Most likely I was intending to make sure temp file is flushed to disk before, but most likely it is a better idea to add `delete=False` to the temp file creation, keep the file name in a variable and run the command out the context manager

Reply:
This is te right thing to do. Outside the context manager, the file is gone.
Alternativelly, you can use `tempfile.mkdtemp()` to hae a new uniqe directory on `/tmp` and then create a file there with a stable name, deferring the cleanup for OS reboot. I like the current approach better :)

Reply:
Is it guaranteed that promtool is going to gete properly written? That my main doubt, to make sure nothing gets buffered before passing the file to the command

Reply:
`fp.flush()` to make sure

Reply:
fixed in c4d9dab